pkg/diffs/diffs.go

```diff
@@ -2,30 +2,68 @@ package diffs
 
 import (
 	"fmt"
-	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
 
 	"k8s.io/apimachinery/pkg/api/equality"
+	utildiff "k8s.io/apimachinery/pkg/util/diff"
 	prowconfig "k8s.io/test-infra/prow/config"
+
+	"github.com/openshift/ci-operator-prowgen/pkg/rehearse"
 )
 
-const configInRepoPath = "cluster/ci/config/prow/config.yaml"
-const jobConfigInRepoPath = "ci-operator/jobs"
+const (
+	ConfigInRepoPath    = "cluster/ci/config/prow/config.yaml"
+	JobConfigInRepoPath = "ci-operator/jobs"
+
+	logRepo    = "repo"
+	logJobName = "job-name"
+	logDiffs   = "diffs"
+
+	objectContainers = ".Spec.Containers"
+	objectVolumes    = ".Spec.Volumes"
+	objectAgent      = ".Agent"
+
+	chosenJob = "Chosen for rehearsal"
+)
 
 // GetChangedPresubmits returns a mapping of repo to presubmits to execute.
-func GetChangedPresubmits(prowMasterConfig *prowconfig.Config, candidateRepoPath string) (map[string][]prowconfig.Presubmit, error) {
+func GetChangedPresubmits(prowMasterConfig, prowPRConfig *prowconfig.Config, logger logrus.FieldLogger) (map[string][]prowconfig.Presubmit, error) {
 	ret := make(map[string][]prowconfig.Presubmit)
 
-	candidateConfigPath := filepath.Join(candidateRepoPath, configInRepoPath)
-	candidateJobConfigPath := filepath.Join(candidateRepoPath, jobConfigInRepoPath)
-	prowPRConfig, err := prowconfig.Load(candidateConfigPath, candidateJobConfigPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load PR's Prow config: %v", err)
-	}
-
 	masterJobs := getJobsByRepoAndName(prowMasterConfig.JobConfig.Presubmits)
 	for repo, jobs := range prowPRConfig.JobConfig.Presubmits {
 		for _, job := range jobs {
-			if !equality.Semantic.DeepEqual(masterJobs[repo][job.Name].Spec, job.Spec) {
+			masterJob := masterJobs[repo][job.Name]
+			logFields := logrus.Fields{logRepo: repo, logJobName: job.Name}
+			chosen := false
+
+			// If the agent was changed, just choose the job for rehearse.
+			if masterJob.Agent != job.Agent {
+				logFields[logDiffs] = convertToReadableDiff(masterJob.Agent, job.Agent, objectAgent)
+				logger.WithFields(logFields).Info(chosenJob)
+				ret[repo] = append(ret[repo], job)
+				continue
+			}
+
+			switch job.Agent {
+			case rehearse.JenkinsAgent:
+				// TODO: safely compare jenkins jobs
```

Comment:
I think we can always skip it

Reply:
IMHO I would prefer at least output a log to the user and tell him that we don't rehearse it.  Or we can just skip it and include this information in our docs.

Reply:
Outputting a log sounds like a good idea to me